modules/dynamic-operator/api/v1alpha1/kuberlogicservice_webhook.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"github.com/kuberlogic/kuberlogic/modules/dynamic-operator/plugin/commons"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var log = ctrl.Log.WithName("kuberlogicservice-webhook")
+
+var pluginInstances map[string]commons.PluginService
+
+func (r *KuberLogicService) SetupWebhookWithManager(mgr ctrl.Manager, plugins map[string]commons.PluginService) error {
+	pluginInstances = plugins
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-kuberlogic-com-v1alpha1-kuberlogicservice,mutating=true,failurePolicy=fail,sideEffects=None,groups=kuberlogic.com,resources=kuberlogicservices,verbs=create;update,versions=v1alpha1,name=mkuberlogicservice.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &KuberLogicService{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *KuberLogicService) Default() {
+	log.Info("default", "name", r.Name)
+
+	plugin, ok := pluginInstances[r.Spec.Type]
+	if !ok {
+		log.Info("Plugin is not loaded", "type", r.Spec.Type)
+		return
+	}
+
+	resp := plugin.Default()
+	if resp.Error != "" {
+		err := errors.New(resp.Error)
+		log.Error(err, "error rpc call 'Default'")
+		return
+	}
+	if r.Spec.Replicas == 0 {
```

Comment:
- 0 replicas can be valid request (suspend behaviour). Instead we should compare the value with nil
- nil is unsupported value for int32 so it's not necessary to check the nil value.

Reply:
Discussed: remove checking zero value